chef/cookbooks/monasca/recipes/agent.rb

```diff
@@ -14,38 +14,111 @@
 # limitations under the License.
 #
 
-### TODO: uncomment this once there is a package.
-# package "openstack-monasca-agent"
+package "openstack-monasca-agent"
 
-### FIXME: remove this once there is a package creating this directory
+agent_settings = node[:monasca][:agent]
 
-directory "/etc/monasca/agent/" do
-  owner "root"
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+monasca_servers = search(:node, "roles:monasca-server") || []
+monasca_server = monasca_servers[0]
+
+ha_enabled = node[:monasca][:ha][:enabled]
+ssl_enabled = node[:monasca][:api][:ssl]
+monasca_protocol = ssl_enabled ? "https" : "http"
+monasca_public_host = CrowbarHelper.get_host_for_public_url(
+  monasca_server, ssl_enabled, ha_enabled
+)
+
+agent_dimensions = { service: "monitoring" }
+service = "monitoring"
+monasca_url = "#{monasca_protocol}://#{monasca_public_host}/v3"
+elasticsearch_url = "http://localhost:9200/"
+kibana_url = "http://#{monasca_public_host}:5601"
+log_api_url = "http://#{monasca_public_host}:#{node[:monasca][:log_api][:bind_port]}"
+
+# template "/etc/monasca/agent/agent.yaml" do
+#   source "agent.yaml.erb"
+#   owner agent_settings["user"]
+#   group agent_settings["group"]
+#   mode 0o640
+#   variables(
+#     monasca_url: monasca_url,
+#     agent_settings: agent_settings,
+#     keystone_settings: keystone_settings,
+#     agent_dimensions: agent_dimensions
+#   )
+#   ### FIXME: Uncomment once we have a package that creates a monasca-agent service
+#   # notifies :reload, resources(service: "openstack-monasca-agent")
+# end
+
+template "/etc/sudoers.d/monasca-agent_sudoers" do
+  source "monasca-agent_sudoers.erb"
+  mode "0440"
   group "root"
-  mode 0o755
-  recursive true
-  notifies :create, "template[/etc/monasca/agent/agent.yaml]"
+  owner "root"
+  variables(
+    agent_user: agent_settings["user"]
+  )
 end
 
-### TODO:
-# * populate agent_settings with useful values for monasca_url and keystone
-#   settings
-# * generate per-node keystone credentials in
-#   crowbar_framework/app/models/monasca_service.rb
-# * use credentials generated in the previous step to create a keystone account
-#   here (using keystone_register)
+# Get the Chef::CookbookVersion for this cookbook
+cb = run_context.cookbook_collection[@cookbook_name]
 
-agent_settings = node[:monasca][:agent][:config]
+cb.manifest["templates"].each do |cbt|
```

Comment:
I think that we are making a mistake here.

Basically hardcoding stuff you with to monitor (processes, http checks and so on) makes an assumption that you those applications, services etc will be always there. Which is basically not true.

I am afraid we cannot copy ansible-monasca-agent directly 1:1 here because there's no clear concept of host_vars + deployment is somehow dynamically mastered from UI (even if our own installer follows different principle right now). So I would drop all of those templates and rely purely on **full auto-detection** here and if we managed to find an app or whatever that is not supported in anyway we can always either:

* extend existing plugin of monasca-agent
* write new plugin